module/User/src/User/Authentication/Adapter/PinMapper.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace User\Authentication\Adapter;
+
+use Zend\Authentication\Adapter\AdapterInterface,
+    Zend\Authentication\Result,
+    User\Mapper\User as UserMapper,
+    User\Model\User as UserModel,
+    User\Model\UserRole as UserRoleModel;
+
+class PinMapper implements AdapterInterface
+{
+
+    /**
+     * Mapper.
+     *
+     * @var UserMapper
+     */
+    protected $mapper;
+
+    /**
+     * Legacy service.
+     *
+     * @var \Application\Service\Legacy
+     */
+    protected $legacyService;
+
+    /**
+     * Lidnr.
+     *
+     * @var string
+     */
+    protected $lidnr;
+
+    /**
+     * Pincode.
+     *
+     * @var string
+     */
+    protected $pincode;
+
+
+    /**
+     * Constructor.
+     *
+     * @param \Application\Service\Legacy
+     */
+    public function __construct($legacyService)
+    {
+        $this->legacyService = $legacyService;
+    }
+
+    /**
+     * Try to authenticate.
+     *
+     * @return Result
+     */
+    public function authenticate()
+    {
+        $mapper = $this->getMapper();
+
+        $user = $mapper->findByLogin($this->lidnr);
+
+        if (null === $user) {
+            return new Result(
+                Result::FAILURE_IDENTITY_NOT_FOUND,
+                null,
+                []
+            );
+        }
+
+        if (!$this->verifyPincode($user)) {
+            return new Result(
+                Result::FAILURE_CREDENTIAL_INVALID,
+                null,
+                []
+            );
+        }
+
+        /**
+         * Users logging in in this way should not have all their regular roles. Since this login
+         * method is less secure.
+         */
+        $userRole = new UserRoleModel();
+        $userRole->setRole('sosuser');
+        $userRole->setLidnr($this->lidnr);
+        $user->setRoles([$userRole]);
+
```

Comment:
Hier moet nog even naar gekeken worden, want dit is geen goede oplossing. Bijvoorbeeld door de IsBoardMember() assertion kan iemand nog steeds alle rechten krijgen.